misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java

```diff
@@ -72,34 +93,58 @@
                             "too many arguments in call to '" + functionName + "'."));
                 }
 
-                String parameterName = params.get(i).getName().get();
-                Value argValue = arg.getValue().evaluate().getJdiValue();
-                // For the ballerina function parameters with type "any", the generated runtime method will only accept
-                // the args which are subtypes of "java.lang.Object". Therefore all the primitive typed arguments
-                // must be converted into their wrapper implementations (i.e. 'int' -> 'Integer'), before passing
-                // into the method.
-                if (params.get(i).typeDescriptor().typeKind().getName().equals(BVariableType.ANY.getString())) {
-                    argValue = getValueAsObject(context, argValue);
+                BExpressionValue argValue = arg.getValue().evaluate();
+                Value jdiArgValue = argValue.getJdiValue();
+                if (params.get(paramIndex).typeDescriptor().signature().equals(BVariableType.ANY.getString())) {
+                    jdiArgValue = getValueAsObject(context, jdiArgValue);
+                }
+
+                if (params.get(paramIndex).paramKind() == ParameterKind.REST) {
+                    for (Map.Entry<String, ParameterSymbol> entry : remainingParams.entrySet()) {
+                        ParameterKind parameterType = entry.getValue().paramKind();
+                        if (parameterType == ParameterKind.REST) {
+                            restParamName = entry.getKey();
+                            restParamTypeName = entry.getValue().typeDescriptor().signature();
+                            break;
+                        }
+                    }
+                    if (restParamName == null) {
+                        throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
+                                "undefined rest parameter."));
+                    }
+
+                    if (!restArgsFound) {
+                        restArrayType = resolveType(context, restParamTypeName);
+                        restArgsFound = true;
+                    }
+                    Value elementType = getElementType(context, restArrayType);
+                    if (!checkIsType(context, jdiArgValue, elementType)) {
+                        throw new EvaluationException(String.format(EvaluationExceptionKind.TYPE_MISMATCH.getString(),
+                                restParamTypeName.replaceAll(BallerinaTypeResolver.ARRAY_TYPE_SUFFIX, ""),
+                                argValue.getType().getString(), restParamName));
+                    }
+                    restValues.add(jdiArgValue);
+                    remainingParams.remove(restParamName);
+                } else {
+                    String parameterName = params.get(paramIndex).getName().get();
```

Comment:
Using `Optional.get()` without `isPresent()` check. Shall we add the necessary checks here?

Reply:
Fixed in f0ae64f